scripts/start-fed.sh

```diff
@@ -9,6 +9,5 @@ SKIPPED_SERVERS=${SKIPPED_SERVERS:-0}
 cd $CFG_DIR
 for ((ID=SKIPPED_SERVERS; ID<FED_SIZE; ID++)); do
   echo "starting mint $ID"
-  ($BIN_DIR/server $CFG_DIR/server-$ID.json 2>&1 | sed -e "s/^/mint $ID: /" ) &
-  echo $! >> $PID_FILE
+  ( ($BIN_DIR/server $CFG_DIR/server-$ID.json 2>&1 & echo $! >&3 ) 3>>$PID_FILE | sed -e "s/^/mint $ID: /" ) &
```

Comment:
I see, so this puts the PIDs at the end of the file after the server process ends?

Reply:
I think the braces `( cmd1 > 1 & cmd2 >3) | sed uses 1` make the commands "share" the stdout and sed just cares for 1

Reply:
What was the stackoverflow post?

The server should be redirecting std error to std output with `2>&1` then run the background, which makes sense since the server still outputs while running.  I expected something like `2>/dev/null` if it was suppressing the error messages.  Could it be something like it outputs the PID in the background now so the server PIDs end up at the end of the PID_FILE?

Reply:
My suggestion is let's just write `server`, `lightningd` and `bitcoind` PIDs to different PID files then close them in that order.

Reply:
[this](https://stackoverflow.com/questions/1652680/how-to-get-the-pid-of-a-process-that-is-piped-to-another-process-in-bash). I'm not sure if I understand what you mean. But yes that sounds like a good idea (can push this in abt 4hrs have to go now)

Reply:
So just write the `bitcoind` PID to `$TMP_DIR/.pid_bitcoind` etc.  Then in the `kill_minimint_processes` function we can read from the PID files in the order we want the services killed, which should be server, lightningd then bitcoind

Reply:
Ah NVM I was thinking this was related to a different issue.